src/Plutarch/Test/QuickCheck/Internal.hs

```diff
@@ -701,13 +714,37 @@ instance PArbitrary PCredential where
       , pconT $ PPubKeyCredential $ pdcons @"_0" # pdata pk # pdnil
       ]
 
--- | @since 2.0.0
+-- | @since 2.2.2
+instance PArbitrary PTxId where
+  parbitrary = do
+    -- PTxId should be 32 bytes long
+    bs <- genByteString 32
+    pure . pconT $ PTxId $ pdcons @"_0" # pdata (pconstant bs) # pdnil
+
+-- | @since 2.2.2
+instance PArbitrary PTxOutRef where
+  parbitrary = do
+    TestableTerm id' <- parbitrary
+    TestableTerm idx <- abs <$> parbitrary
+    pure . pconT $ PTxOutRef $ pdcons # pdata id' #$ pdcons # pdata idx # pdnil
+  pshrink ref = do
+    let (TestableTerm ref') = ref
```

Comment:
This can be matched on function definition isn't it?

Or is it a way to appease weird typing system issue?

Reply:
like
```hs
pshrink (TestableTerm ref') = do
...
```
dropping `let` statement

Reply:
It's probably possible. I'll try it and see.

Reply:
Seems like you can! Fixed and pushed.